src/librustc_ast_passes/ast_validation.rs

```diff
@@ -883,30 +886,36 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                 if is_auto == IsAuto::Yes {
                     // Auto traits cannot have generics, super traits nor contain items.
                     if !generics.params.is_empty() {
+                        let spans: Vec<_> = generics.params.iter().map(|i| i.ident.span).collect();
                         struct_span_err!(
                             self.session,
-                            item.span,
+                            spans,
                             E0567,
                             "auto traits cannot have generic parameters"
                         )
+                        .span_label(item.ident.span, "auto trait cannot have generic parameters")
                         .emit();
                     }
                     if !bounds.is_empty() {
+                        let spans: Vec<_> = bounds.iter().map(|b| b.span()).collect();
```

Comment:
Same as above -- would like to have one larger span instead as well as a label (plus you can add a suggestion if you fancy).